src/OpenLoco/Map/BuildingTile.cpp

```diff
@@ -1,24 +1,58 @@
+#include "../CompanyManager.h"
 #include "../Game.h"
 #include "../GameCommands/GameCommands.h"
 #include "../Objects/BuildingObject.h"
 #include "../StationManager.h"
 #include "../TownManager.h"
+#include "../Ui/WindowManager.h"
 #include "../ViewportManager.h"
 #include "AnimationManager.h"
 #include "Tile.h"
 #include "TileManager.h"
 
 namespace OpenLoco::Map
 {
-    static void sub_497DC1(const Map::Pos2& loc, uint32_t population, uint32_t unk1, uint16_t rating, uint16_t unk3)
+    static loco_global<Town*, 0x01135C38> dword_1135C38;
+
+    // 0x00497DC1
+    // The return value of this function is also being returned via dword_1135C38.
+    static Town* sub_497DC1(const Map::Pos2& loc, uint32_t population, uint32_t unk1, uint16_t rating, uint16_t unk3)
     {
-        registers regs;
-        regs.edi = population;
-        regs.esi = unk1;
-        regs.ebp = rating | (unk3 << 16);
-        regs.ax = loc.x;
-        regs.cx = loc.y;
-        call(0x00497DC1, regs);
+        auto res = TownManager::getClosestTownAndUnk(loc);
+        if (res == std::nullopt)
+        {
+            dword_1135C38 = nullptr;
```

Comment:
If you plan to hook this function keep in mind this will need to be changed back to 0xFFFFFFFF instead of nullptr